### PR TITLE
#102 Change the addition of a block to another

### PR DESCRIPTION
**Pull request by fkroockmann**
Actually the addition of one block is too bound to dnd.

### CODE REVIEW

File: src/tb/apps/content/components/DndManager.js [hunk 1]
```diff
@@ -144,70 +142,63 @@ define(
...
-                        parent = content.jQueryObject.parent('.' + this.contentClass);
-                        this.dropAsHtml(config, content.jQueryObject);
 
-                        if (parent.data(this.idDataAttribute)) {
-                            parentAsContent = ContentContainer.find(parent.data(this.idDataAttribute));
-                        } else {
-                            parentAsContent = ContentManager.buildElement(parent);
-                        }
-                        parentAsContent.setUpdated(true);
+                        oldParent = content.jQueryObject.parent('.' + this.contentClass);
```
Comment (ndufreche) [inline src/tb/apps/content/components/DndManager.js]: Refering to the jQuery documentation :
The .parents() and .parent() methods are similar, except that the latter only travels a single level up the DOM tree.

Are you sure that line work in all case ?
What append if the parent you are searching is not on the previous child level but two or three level before ?

File: src/tb/apps/content/components/DndManager.js [hunk 2]
```diff
@@ -279,17 +271,14 @@ define(
 
                     var target = jQuery(event.target),
                         config = {},
-                        prev = target.prev('.' + this.contentClass),
-                        parent = target.parent();
+                        parent = target.parent(this.dropZoneAttribute),
```
Comment (ndufreche) [inline src/tb/apps/content/components/DndManager.js]: Refering to the jQuery documentation :
The .parents() and .parent() methods are similar, except that the latter only travels a single level up the DOM tree.

Are you sure that line work in all case ?
What append if the parent you are searching is not on the previous child level but two or three level before ?

File: src/tb/apps/content/models/ContentSet.js [hunk 3]
```diff
@@ -35,6 +35,46 @@ define(['content.models.AbstractContent', 'jquery', 'jsclass'], function (Abstra
...
+         * If position is not provided, html will be at the first
+         * @param {String} html
+         * @param {Number} position
+         * @returns {Promise}
+         */
+        append: function (content, position) {
+            var self = this,
+                done = false,
+                dfd = jQuery.Deferred(),
+                childrens = this.getNodeChildrens();
```
Comment (ndufreche) [inline src/tb/apps/content/models/ContentSet.js]: Typo fail children not childrens

File: src/tb/apps/content/components/DndManager.js (hunk 1 above)
Comment (fkroockmann) [inline src/tb/apps/content/components/DndManager.js]: You are right the parent method not enough, i will use parents method with :first attribute.

File: src/tb/apps/content/components/DndManager.js [hunk 4]
```diff
@@ -84,7 +82,7 @@ define(
                             contentSet.isChildrenOf(currentContentId);
                             if (contentSet.id !== currentContentId && !contentSet.isChildrenOf(currentContentId)) {
 
-                                childrens = contentSet.getNodeChildrens();
+                                childrens = contentSet.getNodeChildren();
```
Comment (ndufreche) [inline src/tb/apps/content/components/DndManager.js]: It stay a typo fail

**Comment by ndufreche** [comment]
:+1:

**Pull request merged**